mlrun/projects/project.py

```diff
@@ -1889,6 +1889,12 @@ def _enrich_artifact_path_with_workflow_uid(self):
         artifact_path = self.spec.artifact_path or mlrun.mlconf.artifact_path
         if not mlrun.mlconf.enrich_artifact_path_with_workflow_id:
             return artifact_path
+
+        # no need to add workflow.uid to the artifact path for uniqueness, this is already being handled by generating
+        # the artifact target path from the artifact content hash ( body / file etc...)
+        if mlrun.mlconf.should_generate_target_path_from_artifact_hash():
+            return artifact_path
+
         workflow_uid_string = "{{workflow.uid}}"
         if workflow_uid_string in artifact_path:
             return artifact_path
```

Comment:
These 3 if statements can be combined into one
```suggestion
        workflow_uid_string = "{{workflow.uid}}"
        if (
            not mlrun.mlconf.enrich_artifact_path_with_workflow_id or
            
            # no need to add workflow.uid to the artifact path for uniqueness, this is already being handled by generating
            # the artifact target path from the artifact content hash ( body / file etc...)
            mlrun.mlconf.should_generate_target_path_from_artifact_hash() or
            workflow_uid_string in artifact_path
        ):
            return artifact_path
```